test/apps/iodemo/io_demo.cc

```diff
@@ -721,15 +772,34 @@ class DemoServer : public P2pDemoCommon {
         VERBOSE_LOG << "receiving AM IO write data";
         assert(msg->data_size != 0);
 
-        BufferIov *iov             = _data_buffers_pool.get();
-        IoWriteResponseCallback *w = _callback_pool.get();
+        UcxCallback *cb;
+        BufferIov *iov;
 
-        iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
-        w->init(this, conn, msg->sn, iov, &_curr_state.write_count);
+        am_recv_iovs_map_t::iterator iter = _am_recv_iovs.find(msg->sn);
+        if (iter == _am_recv_iovs.end()) {
+            IoWriteResponseCallback *w = _callback_pool.get();
+            iov                        = _data_buffers_pool.get();
+            cb                         = w;
+
+            iov->init(msg->data_size, _data_chunks_pool, msg->sn, opts().validate);
+            w->init(this, conn, msg->sn, iov, &_curr_state.write_count);
+            if (iov->size() > 1) {
+                UcxAmMsg msg_desc(iov, w);
+                _am_recv_iovs[msg->sn] = msg_desc;
```

Comment:
minor: better (for compiler) to do in one line assignment:
```
_am_recv_iovs[msg->sn] = UcxAmMsg(iov, w);
```
same for `handle_io_am_read_reply`

Reply:
ok